### ISSUE TITLE
Which Python, Tensorflow, Keras, CUDA version for GPU usage?

### ISSUE THREAD

**Issue by CarlSeeTwo** [matterport/Mask_RCNN#2312]
I have tried many times in many different Anaconda environments with many different combinations of Python, Tensorflow, Keras, and CUDA to try to get this to run with my GPU (GTX 1060), but I have never been successful.  Always some sort of build error, or import error.

I know how to install CUDA and the appropriate cuDNN packages.

I am on Windows 10 and I want to use an Anaconda environment for setup.

Does anybody have a combination of all of these that they know works today, with the current Mask RCNN?

**Comment by MateRyze** [comment]
Yes, try this fork with Tensorflow 2. It works great with CUDA 10.1
https://www.immersivelimit.com/tutorials/mask-rcnn-for-windows-10-tensorflow-2-cuda-101

**Comment by burhr2** [comment]
Thanks, @MateRyze.  Kindly direct me to a repo which has the evaluation such mean average precision (mAP), MIoU, accuracy, ROC/AUC if you have found one.

**Comment by maskrcnnuser** [comment]
> Yes, try this fork with Tensorflow 2. It works great with CUDA 10.1
> https://www.immersivelimit.com/tutorials/mask-rcnn-for-windows-10-tensorflow-2-cuda-101

Works well with Demo but gives errors while training!

**Comment by haseeb33** [comment]
Used both repo's:

```https://github.com/akTwelve/Mask_RCNN```(based on https://www.immersivelimit.com/tutorials/mask-rcnn-for-windows-10-tensorflow-2-cuda-101)
```https://github.com/leekunhee/Mask_RCNN```
claimed to be working with ```tf2``` but stuck at training(not even showing any errors).

My environment:
    ```CUDA 11.0```
    ```tf 2.3.0```

**Comment by ManuelZ** [comment]
@haseeb33 Same here, although I can train the balloons example.

**Comment by Ademord** [comment]
@ManuelZ @haseeb33 did you modify anything or does it work out of the box for you? I ran the demo notebook and it works but the notebooks in /samples do not work at all. Help pls? Maybe u can share your changes?

**Comment by ManuelZ** [comment]
@Ademord I don't think I modify anything, maybe only the path towards the data when calling the balloons script.
I finally got my script working (https://github.com/ManuelZ/steel-defect-detection/blob/master/train.py), I had messed up the returning of labels in the `load_mask` function.

**Comment by AvalonChang** [comment]
My environment (only list specify version)
Python: 3.6.8
TensorFlow-GPU: 1.13.1
Keras: 2.3.1
h5py: 2.8

GPU: GeForce RTX 2060 SUPER
CUDA Toolkit: 10.0.130
cuDNN: 7.6.0

I can run the demo successfully.


when running the train_shapes, because Keras version is too high.
must modify model.py, line: 2199.
reference: [https://github.com/matterport/Mask_RCNN/issues/2410](url)

then, if training stopped at the first step.
this issue can help [https://github.com/matterport/Mask_RCNN/issues/1962](url)
I update my GPU driver version to 457.51, now I can run the train_shapes successfully too.

**Comment by kimile599** [comment]
> Thanks, @MateRyze. Kindly direct me to a repo which has the evaluation such mean average precision (mAP), MIoU, accuracy, ROC/AUC if you have found one.

Hi，

Do you find some repos on the model evaluation?
Thank you.

**Comment by getch-geohum** [comment]
Hello every one,
I train the model with tf 2.6 and cuda 11.5 using the model from @waleedka and @akTwelve repos on NVIDIA A100 GPU. The speed is superfast, the problem is the prediction outputs are very strange.
![image](https://user-images.githubusercontent.com/82418348/152749503-741b5daa-74d3-4a84-96e5-3135645649cc.png)
Is there any one who have a solution?

**Comment by greatgwy** [comment]
> I have tried many times in many different Anaconda environments with many different combinations of Python, Tensorflow, Keras, and CUDA to try to get this to run with my GPU (GTX 1060), but I have never been successful. Always some sort of build error, or import error.
> 
> I know how to install CUDA and the appropriate cuDNN packages.
> 
> I am on Windows 10 and I want to use an Anaconda environment for setup.
> 
> Does anybody have a combination of all of these that they know works today, with the current Mask RCNN?

I am using python 3.7.9, tf-gpu 1.15, keras 2.1.6, CUDA10.0, h5py 2.10, scipy 1.2.1, scikit-image 0.18.3  on win10.

**Comment by jyothirrose** [comment]
- opencv-python==4.1.0.25
    - tensorflow-gpu==1.15.0
    - cython
    - h5py
    - imgaug==0.3.0
    - keras==2.1.3
    - numpy==1.18.1
    - pillow
    - scikit-image==0.15.0
    - shapely==1.7.0
    
use these packages in requirements.txt or install as u wish in the conda environment. after that conda install cudatoolkit=10.0 and cudnn=7.0. Tested and working on windows 10 PC
